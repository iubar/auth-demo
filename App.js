import * as React from 'react';
import { Text, View } from 'react-native-paper';
import { Provider as PaperProvider } from 'react-native-paper';
import { Ionicons } from '@expo/vector-icons';
import { createMaterialBottomTabNavigator } from '@react-navigation/material-bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';
import PasswordGrant from './src/screens/PasswordGrant';
import AuthorizationCodeGrant from './src/screens/AuthorizationCodeGrant';
import HttpCall from './src/screens/HttpCall';
import Toolbar from './src/components/ToolbarComponent';

 
 const Tab = createMaterialBottomTabNavigator(); 
 
export default class App extends React.Component {	



 
 /*
  const [routes] = React.useState([
    { key: 'password_grant', title: 'Password grant', icon: 'album' },
    { key: 'authorization_code_grant', title: 'Authorization Code grant', icon: 'album' },
    { key: 'http_call', title: 'Http call', icon: 'album' },
  ]); 

  const renderScene = BottomNavigation.SceneMap({
    password_grant: PasswordGrantRoute,
    authorization_code_grant: AuthorizationCodeGrantRoute,
    http_call: HttpCallRoute,
  });
*/



 
	render() {
		

		
	  return (
	  <PaperProvider>
		
		<NavigationContainer>
		<Toolbar />
      <Tab.Navigator
	  	  initialRouteName="AuthorizationCodeGrant"
		  activeColor="#f0edf6"
		  inactiveColor="#3e2465"
		  barStyle={{ backgroundColor: '#694fad' }}		  
        screenOptions={({ route }) => ({
          tabBarIcon: ({ focused, color, size }) => {
            let iconName;

            if (route.name === 'Password Grant') {
              iconName = focused ? 'ios-information-circle' : 'ios-information-circle-outline';
            } else if (route.name === 'Auth Code Grant') {
              iconName = focused ? 'ios-list-box' : 'ios-list';
            } else if (route.name === 'Api call') {
              iconName = focused ? 'ios-list-box' : 'ios-list';
            }

            // You can return any component that you like here!
            return <Ionicons name={iconName} size={size} color={color} />;
          },
        })}
        tabBarOptions={{
          activeTintColor: 'tomato',
          inactiveTintColor: 'gray',
        }}
      >
 
			<Tab.Screen name="Password Grant" component={PasswordGrant} />
			<Tab.Screen name="Auth Code Grant" component={AuthorizationCodeGrant} />
			<Tab.Screen name="Api call" component={HttpCall} />
		  </Tab.Navigator>
		</NavigationContainer>		
		</PaperProvider>
	  );
	}
  
  
};

 