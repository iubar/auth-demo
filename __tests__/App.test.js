import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import fetchMock from 'jest-fetch-mock';

fetchMock.enableMocks();

global.fetch = jest.fn(() =>
	Promise.resolve({
		json: () => Promise.resolve({ status: 400, error: 'mocked', data: [], response: '' }),
	})
);

beforeEach(() => {
	fetchMock.mockClear();
	fetchMock.mockReject(() => Promise.reject('API is down'));
});

describe('<App />', () => {
	it('has 1 child', () => {
		const tree = renderer.create(<App />).toJSON();
		expect(tree.children.length).toBe(1);
	});
	it('matches snapshot and renders correctly', () => {
		let tree = renderer.create(<App />);
		expect(tree).toMatchSnapshot();
	});
});
