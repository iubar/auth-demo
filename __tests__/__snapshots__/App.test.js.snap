// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(28, 28, 30)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          Auth Demo
        </Text>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(28, 28, 30)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 11,
                },
                Object {
                  "color": "rgba(255, 255, 255, 0.7)",
                },
                undefined,
              ],
            ]
          }
        >
          iubar.it
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          collapsable={false}
          importantForAccessibility="auto"
          pointerEvents="auto"
          removeClippedSubviews={false}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "display": "flex",
              },
            ]
          }
          testID="RouteScreen: 1"
        >
          <View
            collapsable={false}
            style={
              Object {
                "flex": 1,
                "top": 0,
              }
            }
          >
            <RCTSafeAreaView
              emulateUnlessSupported={true}
            >
              <RCTScrollView
                style={
                  Object {
                    "paddingHorizontal": 20,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "rgb(28, 28, 30)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "textAlign": "left",
                          },
                          Object {
                            "color": "rgba(28, 28, 30, 0.87)",
                            "fontFamily": "System",
                            "fontWeight": "500",
                            "writingDirection": "ltr",
                          },
                          Array [
                            Object {
                              "fontSize": 20,
                              "letterSpacing": 0.15,
                              "lineHeight": 30,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    Auth Code Grant with PKCE
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "rgb(28, 28, 30)",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "textAlign": "left",
                          },
                          Object {
                            "color": "rgba(28, 28, 30, 0.87)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                            "writingDirection": "ltr",
                          },
                          Array [
                            Object {
                              "fontSize": 14,
                              "letterSpacing": 0.25,
                              "lineHeight": 20,
                              "marginVertical": 2,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    Client Id: 
                    0
                  </Text>
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "rgb(28, 28, 30)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          undefined,
                        ]
                      }
                    >
                      That client doesn't support the Auth Code Grant flow
                    </Text>
                  </View>
                </View>
              </RCTScrollView>
            </RCTSafeAreaView>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "bottom": 0,
            "elevation": 4,
            "left": 0,
            "position": null,
            "right": 0,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 0, 0, 1)",
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="tablist"
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "marginBottom": 0,
                  "marginHorizontal": 0,
                  "maxWidth": 384,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": undefined,
                  "borderRadius": 0,
                  "height": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "scale": 0.008,
                    },
                  ],
                  "width": 0,
                }
              }
            />
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 7,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "rgb(255, 69, 58)",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 8,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 0.5,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(28, 28, 30)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "height": 56,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      Password Grant
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": true,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "rgb(255, 69, 58)",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 8,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(28, 28, 30)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "height": 56,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      Auth Code Grant
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 7,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "rgb(255, 69, 58)",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 8,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 0.5,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(28, 28, 30)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "height": 56,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      Api
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "selected": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 6,
                }
              }
            >
              <View
                pointerEvents="none"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 24,
                      "marginHorizontal": 12,
                      "marginTop": 2,
                      "transform": Array [
                        Object {
                          "translateY": 7,
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "left": 0,
                          "position": "absolute",
                          "top": -2,
                        },
                        Object {
                          "right": -2,
                        },
                      ]
                    }
                  >
                    <Text
                      collapsable={false}
                      numberOfLines={1}
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "backgroundColor": "rgb(255, 69, 58)",
                          "borderRadius": 8,
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontSize": 8,
                          "fontWeight": "400",
                          "height": 16,
                          "lineHeight": 8,
                          "minWidth": 16,
                          "opacity": 0,
                          "overflow": "hidden",
                          "paddingHorizontal": 4,
                          "textAlign": "center",
                          "textAlignVertical": "center",
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 16,
                      "paddingBottom": 2,
                      "transform": Array [
                        Object {
                          "scale": 0.5,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(28, 28, 30)",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontSize": 12,
                              "height": 56,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#ffffff",
                            },
                          ],
                        ]
                      }
                    >
                      Settings
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
